tests/integration/psync2-master-restart.tcl

```diff
@@ -41,9 +41,12 @@ start_server {} {
         set offset [status $master master_repl_offset]
         $replica config resetstat
 
+        set rd [redis_deferring_client]
```

Comment:
I would rather this be a feature of the re restart_server proc

Reply:
OK, I can do that, but first I want to show another possible approach: taking the offset when shutdown is ongoing and GETACK has been sent. In this way, we get the final offset before the master exits without using NOW.

```Tcl
        set rd [redis_deferring_client]
        $rd shutdown
        set offset [status $master master_repl_offset]
        catch {
            restart_server 0 true false
            set master [srv 0 client]
            
        }
```

:+1: or :-1: ?

Reply:
i feel it's awkward that the test does both SHUTDOWN and restart_server.
i.e. we rely on the fact restart_server will be ok with it.
i'd rather restart_server will take an optional argument indicating how to shutdown.

the other thing that bothers me the risk of the extra GETACK somehow messing up the ability of the replica to PSYNC.
i know that today we avoid sending the GETACK if the replica is already in sync, so all of that won't matter, but i still think it's a good idea to use one shutdown approach in one test and the other approach in the other test (since we have two of them anyway)